clippy_lints/src/struct_fields.rs

```diff
@@ -0,0 +1,168 @@
+use clippy_utils::diagnostics::{span_lint_and_help, span_lint_hir};
+use clippy_utils::str_utils::{count_match_end, count_match_start, to_snake_case};
+use rustc_hir::{FieldDef, ItemKind, VariantData};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::Span;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Detects struct fields that are prefixed or suffixed
+    /// by the same characters.
+    ///
+    /// ### Why is this bad?
+    /// Information common to all struct fields is better represented in the struct name.
+    ///
+    /// ### Limitations
+    /// Characters with no casing will be considered when comparing prefixes/suffixes
+    /// This applies to numbers and non-ascii characters without casing
+    /// e.g. `Foo1` and `foo2` is considered to have different prefixes
+    /// (the prefixes are `foo1` and `foo2` respectively), as also `bar螃`, `bar蟹`
+    ///
+    /// ### Example
+    /// ```rust
+    /// struct Cake {
+    ///     cake_sugar: u8,
+    ///     cake_flour: u8,
+    ///     cake_eggs: u8
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// struct Cake {
+    ///     sugar: u8,
+    ///     flour: u8,
+    ///     eggs: u8
+    /// }
+    /// ```
+    #[clippy::version = "1.74.0"]
+    pub STRUCT_FIELD_NAMES,
+    style,
+    "structs where all fields share a prefix/postfix"
+}
+
+pub struct StructFields {
+    threshold: u64,
+    avoid_breaking_exported_api: bool,
+}
+
+impl StructFields {
+    pub fn new(threshold: u64, avoid_breaking_exported_api: bool) -> StructFields {
+        StructFields {
+            threshold,
+            avoid_breaking_exported_api,
+        }
+    }
+}
+
+impl_lint_pass!(StructFields => [STRUCT_FIELD_NAMES]);
+
+fn is_snake_case(name: &str) -> bool {
+    name.chars().all(|c| !c.is_uppercase() || c == '_')
+}
+
+fn check_fields(cx: &LateContext<'_>, threshold: u64, fields: &[FieldDef<'_>], item_name: &str, item_span: Span) {
+    if (fields.len() as u64) < threshold {
+        return;
+    }
```

Comment:
Yes, it is expected, I've implemented the threshold imitating the behavior of the enum_variants lint and the reasoning given in the issue that generated the option (that in enums with low variant counts the change doesn't improve the quality of the names that much) https://github.com/rust-lang/rust-clippy/issues/1138. 
I think it makes sense to make the threshold affect at the lint as a whole. Otherwise it might make sense to have two independent lints.